android/app/src/main/java/app/organicmaps/util/log/Logger.java

```diff
@@ -88,16 +88,17 @@ public static void e(String tag, String msg, Throwable tr)
     log(Log.ERROR, tag, msg, tr);
   }
 
-  // Index of stacktrace depth where the original log method call resides.
-  private static final int CALL_STACK_INDEX = 3;
-
   @NonNull
   private static String getSourcePoint()
   {
     final StackTraceElement[] stackTrace = new Throwable().getStackTrace();
-    assert stackTrace.length > CALL_STACK_INDEX : "Synthetic stacktrace doesn't have enough elements";
-
-    final StackTraceElement st = stackTrace[CALL_STACK_INDEX];
+    // Skip the chain of Logger.x() -> Logger.log() calls.
+    int f = 0;
+    for (; f < stackTrace.length && stackTrace[f].getClassName().equals(Logger.class.getName()); f++);
+    // The stack trace should have at least one non-logger frame, but who wants to crash here if it doesn't?
```

Comment:
what about logging from the Logger?
maybe better to locate the specific call that should always precede log's caller? is it `Logger.log()`?

another approach might be to disable r8 / add @keep to log methods to guarantee a stable stacktrace "prefix"